beater/server.go

```diff
@@ -54,18 +54,20 @@ func run(server *http.Server, ssl *SSLConfig) error {
 	}
 }
 
-func stop(server *http.Server, timeout time.Duration) {
+func stop(server *http.Server, timeout time.Duration) (err error) {
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 
-	err := server.Shutdown(ctx)
+	err = server.Shutdown(ctx)
 	if err != nil {
-		logp.Err(err.Error())
-		err = server.Close()
-		if err != nil {
-			logp.Err(err.Error())
-		}
+		defer func() {
```

Comment:
This is a lot problems just to log an error in the caller instead of the callee, which makes no difference at all.

Reply:
It makes a difference for testing. You can't test if something was logged, but you can test if something returned an error.

Reply:
I see no test taking advantage of that difference here. 
If you can come up with a test that exercises all the code paths (graceful shutdown ok, graceful shutdown timeout, and server close error), then the patch is welcome.

Reply:
We don't have these tests yet. But in general I prefer to have logging in as few places as possible and let errors bubble up instead of logging them out.